vnext/Microsoft.ReactNative/ReactNativeWindow.idl

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import "ReactNativeHost.idl";
+import "ReactNativeIsland.idl";
+
+namespace Microsoft.ReactNative
+{
+#ifdef USE_WINUI3
+    struct ReactApplicationInstanceSettings
+    {
+        String bundleRootPath;
+    };
+
+    runtimeclass ReactNativeWindow
```

Comment:
The behavior of this object seems more aligned with the older CoreApp API, to try to simplify creating an initial RNW app.

I dont think we should just call this RNWindow though.  There are other scenarios where you might want to create an additional window hosting RN content within a app which feels more like what I'd expect this object to be doing.  That wouldn't create a new host though, it would attach to an existing host.

Reply:
Can you please provide more context with examples here @acoates-ms ?

Reply:
We have a couple of scenarios internal to RNW that do this - RedBox and Modal for example.   But also any multi window application may want to create additional top level windows all within the same ReactHost.

Reply:
Whatever it is, we expect this to be the one object a template creates, so given `class ReactNativeWindow : WinAppSDK::AppWindow`, the RNW app's code is something so minimal like:

```cpp
int WinMain()
{
    auto rnw= winrt::Microsoft::ReactNative::ReactNativeAppWindow::Create();
    rnw.InstanceSettings().BundleUrl("index");
    // more instance settings
    RegisterAutolinkedNativeModules(rnw.InstanceSettings().PackageProvider());
    rnw.Title(windowTitle);
    rnw.Resize({1000, 1000});
    rnw.Show();
}
```